src/components/popup/details/purposes/purposes.jsx

```diff
@@ -222,17 +247,40 @@ export default class Purposes extends Component {
 																		   index={index}
 																		   purpose={purpose}
 																		   isActive={purposeIsActive(purpose)}
+																		   isLegitimateInterestActive={isPurposeLegitimateInterestActive(index)}
 																		   isTechnical={false}
 																		   createOnShowVendors={this.createOnShowVendors.bind(this)}
+																		   onToggleLegitInterest={this.handleSelectLegitInterest.bind(this)}
 																		   onToggle={this.createHandleSelectPurpose()}/>)}
 							</div>
+							{specialPurposes && !!specialPurposes.length && <div>
+								<LocalLabel className={style.header} prefix="specialPurposes" localizeKey={`title`}>Special purposes</LocalLabel>
+								{specialPurposes.map((purpose, index) => <Purpose key={index}
+																		   index={index}
+																		   purpose={purpose}
+																		   isActive={false}
+																		   isLegitimateInterestActive={false}
+																		   isTechnical={true}
+																		   specialPurpose={true}
+																		   createOnShowVendors={this.createOnShowVendors.bind(this)}
+																		   onToggleLegitInterest={this.handleSelectLegitInterest.bind(this)}
+																		   onToggle={this.createHandleSelectPurpose()}/>)}
+							</div>}
 							<div>
 								<LocalLabel className={style.header} prefix="features" localizeKey={`title`}>Features</LocalLabel>
 								{features.map((feature, index) => <Feature key={index}
 																		   feature={feature}
 																		   createOnShowVendors={this.createOnShowVendors.bind(this)}/>
 								)}
 							</div>
+							{specialFeatures && !!specialFeatures.length && <div>
+								<LocalLabel className={style.header} prefix="specialFeatures" localizeKey={`title`}>Special features</LocalLabel>
+								{specialFeatures.map((feature, index) => <Feature key={index}
+																		   feature={feature}
+																		   specialFeature={true}
```

Comment:
“Special Feature” means one of the features of processing personal data used by
participants in the Framework that are defined in the Policies or the Specifications used in pursuit of one or several Purposes for which the user is given the choice to opt-in separately from the choice afforded regarding the Purposes which they support.

- [ ]  special features should be consentable